src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,666 @@
+                    if (ObjectUtils.isNotNull(subObjectCode)) {
+                        transactionDTO.setSubObjectCodeNumber(subObjectCode.getFinancialSubObjectCode());
+                    } else {
+                        LOG.info("updateTransactionDTOObjectCodes() Invalid Sub Object Code.");
+                        transactionDTO.setTransactionInputError(true);
+                    }
+                }
+            } else {
+                LOG.info("updateTransactionDTOObjectCodes() Invalid Object Code.");
+                transactionDTO.setTransactionInputError(true);
+            }
+        } else {
+            transactionDTO.setObjectCodeNumber(getTransactionObjectCode());
+        }
+    }
+    
+    private void updateTransactionDTOProjectCodeNumber(AmazonBillingDistributionOfIncomeTransactionDTO transactionDTO, AmazonBillingCostCenterDTO costCenterDTO) {
+        if (StringUtils.isNotBlank(costCenterDTO.getProjectCode())) {
+            ProjectCode projectCode = getProjectCodeService().getByPrimaryId(costCenterDTO.getProjectCode());
+            if (ObjectUtils.isNotNull(projectCode)) {
+                transactionDTO.setProjectCodeNumber(projectCode.getCode());
+            } else {
+                LOG.info("updateTransactionDTOProjectCodeNumber() Invalid Project Code.");
+                transactionDTO.setTransactionInputError(true);
+            }
+        }
+    }
+    
+    private boolean shouldBuildDistributionOfIncomeDocument(AmazonAccountDetail amazonAccountDetail, KualiDecimal cost) {
+        for (DocumentHeader dh : findDocumentHeadersForAmazonDetail(amazonAccountDetail)) {
+            try {
+                DistributionOfIncomeAndExpenseDocument di = (DistributionOfIncomeAndExpenseDocument) getDocumentService().getByDocumentHeaderId(dh.getDocumentNumber());
+                if (di.getTotalDollarAmount().equals(cost)) {
+                    LOG.info("shouldBuildDistributionOfIncomeDocument() DI document number " + di.getDocumentNumber() + " already exists for Amazon account: " + amazonAccountDetail.toString());
+                    return false;
+                }
+            } catch (WorkflowException e) {
+                LOG.error("shouldBuildDistributionOfIncomeDocument() There was a problems creating a DI document from a document header: " + dh.getDocumentNumber(), e); 
+                throw new RuntimeException(e);
+            }
+        }
+        return true;
+    }
+    
+    private Collection<DocumentHeader> findDocumentHeadersForAmazonDetail(AmazonAccountDetail amazonAccountDetail) {
+        Map fieldValues = new HashMap();
+        fieldValues.put(CUKFSConstants.DOCUMENT_HEADER_EXPLANATION_FIELD, buildDocumentExplanation(amazonAccountDetail.getAwsAccount()));
+        fieldValues.put(CUKFSConstants.DOCUMENT_HEADER_DESCRIPTION_FIELD, buildDocumentDescription(amazonAccountDetail.getBusinessPurpose()));
```

Comment:
When I switched the code to use the kfspropertyconstants, I get 2016-08-26 11:11:33,316 [scheduler_Worker-2] FATAL org.kuali.rice.core.framework.persistence.jta.KualiTransactionInterceptor :: Exception caught by Transaction Interceptor, this will cause a rollback at the end of the transaction.
org.springframework.jdbc.BadSqlGrammarException: OJB operation; bad SQL grammar []; nested exception is java.sql.SQLSyntaxErrorException: ORA-00904: "DOCUMENTEXPLANATION": invalid identifier
That property doesn't seem to be what we need here.